packages/wdio-shadow-dom-service/src/index.js

```diff
@@ -0,0 +1,29 @@
+export default class ShadowDomService {
+
+    static fnFactory(elementSelector, qsAll) {
+        const strFn = `
+        (function() {
+          // element has a shadowRoot property
+          if (this.shadowRoot) {
+            return this.shadowRoot.querySelector${qsAll ? 'All' : ''}('${elementSelector}')
+          }
+          // fall back to querying the element directly if not
+          return this.querySelector${qsAll ? 'All' : ''}('${elementSelector}')
+        })`
+        return eval(strFn)
+    }
+
+    static async shadow$ (selector) {
```

Comment:
As its built, the user can do browser.$('foo').shadow$('bar').shadow$('bar2'). Have you tested this scenario to see if it works?

Reply:
It should work the same as chaining `$` calls.